rollup-node.md

```diff
@@ -0,0 +1,285 @@
+# Rollup Node Specification
+
+<!-- All glossary references in this file. -->
+[rollup node]: /glossary.md#rollup-node
+[derivation]: /glossary.md#L2-chain-derivation
+[L2 derivation inputs]: /glossary.md#L2-derivation-inputs
+[payload attributes]: /glossary.md#payload-attributes
+[block]: /glossary.md#block
+[execution engine]: /glossary.md#execution-engine
+[reorg]: /glossary.md#re-organization
+[block gossip]: /glossary.md#block-gossip
+[rollup driver]: /glossary.md#rollup-driver
+[deposits]: /glossary.md#deposits
+[deposit-feed]: /glossary.md#L2-deposit-feed-contract
+[L2 chain inception]: /glossary.md#L2-chain-inception
+
+The [rollup node] is the component responsible for [deriving the L2
+chain][derivation] from L1 blocks. This process happens in two steps:
+
+1. Read [L2 derivation inputs] from L1 and generate [payload attributes] (essentially
+   [a block without output properties][block]).
+2. Pass the payload attributes to the [execution engine], so that [output block
+   properties][block] may be computed.
+
+While this process is conceptually a pure function from the L1 chain to the L2
+chain, it is in practice incremental. The L2 chain changes whenever new L1
+blocks (and hence new L2 derivation inputs) are added to the L1 chain, or
+whenever the L1 chain [re-organizes][reorg].
+
+Additionally, the rollup node is responsible for [block gossip], i.e.
+transmitting the L2 blocks it derives over a peer-to-peer network.
+
+The part of the rollup node that derives the L2 chain is called the [rollup
+driver]. This document is mostly concerned with the specification of the rollup
+driver, while block gossip is [specified in a separate document][gossip-spec].
+
+[gossip-spec]: TODO
+
+> **TODO LINK** block gossip spec
+
+## L2 Chain Derivation
+[l2-chain-derivation]: #l2-chain-derivation
+
+This section specifies how the [rollup driver] derives one L2 block from every
```

Comment:
Are we fully committed to doing 1:1 derivation? I thought we would map 1 L1 block to 1 *or more* L2 blocks.

Reply:
It's 1-1 derivation because this is the deposits-only stage. It will change later on when we add sequencer blocks.